From bf8aa0fafd02eefda26c427b17b62aa68c4893ee Mon Sep 17 00:00:00 2001
From: Theo GOUREAU <theo.goureau-ext@st.com>
Date: Wed, 12 Mar 2025 16:47:40 +0100
Subject: [PATCH 821/841] feat(stm32mp2): add SCMI secure shmem in memory map
 for CM33TD

Add SCMI secure shared memory in memory mappings for BL31 in CM33TDCID.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git ea17c9d03ca617bdd15c0d4298d75d3848321b18 feat(stm32mp2): add SCMI secure shmem in memory map for CM33TD]
Change-Id: Iceabbbe48dc65c48ed5b7c70e6ea248a864f35bb
Signed-off-by: Theo GOUREAU <theo.goureau-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/443786
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h     | 10 +++++++++-
 plat/st/stm32mp2/stm32mp2_private.c |  9 +++++++++
 2 files changed, 18 insertions(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index b88e205b6..8b25732ca 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -161,6 +161,12 @@
 #define STM32MP_BSEC_MIRROR_BASE	SRAM1_BASE
 #define STM32MP_BSEC_MIRROR_SIZE	PAGE_SIZE
 
+#if STM32MP_M33_TDCID
+/* The second 4-kByte page of SRAM1 is used in CM33TDCID for the SCMI shared memory  */
+#define STM32MP_SCMI_SEC_SHMEM_BASE	(SRAM1_BASE + PAGE_SIZE)
+#define STM32MP_SCMI_SEC_SHMEM_SIZE	PAGE_SIZE
+#endif /* STM32MP_M33_TDCID */
+
 #define STM32MP_SEC_SYSRAM_BASE		STM32MP_SYSRAM_BASE
 
 #if defined(IMAGE_BL2) && STM32MP_USB_PROGRAMMER && !STM32MP21
@@ -262,7 +268,9 @@ enum ddr_type {
  * MAX_MMAP_REGIONS is usually:
  * BL stm32mp2_mmap size + mmap regions in *_plat_arch_setup
  */
-#if STM32MP_USB_PROGRAMMER || defined(IMAGE_BL31)
+#if defined(IMAGE_BL31) && STM32MP_M33_TDCID
+#define MAX_MMAP_REGIONS			8
+#elif STM32MP_USB_PROGRAMMER || defined(IMAGE_BL31)
 #define MAX_MMAP_REGIONS			7
 #else
 #define MAX_MMAP_REGIONS			6
diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index a4eb7f1dc..3c8966d53 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -64,6 +64,12 @@
 #define MAP_BSEC_MIRROR	MAP_REGION_FLAT(STM32MP_BSEC_MIRROR_BASE, \
 					STM32MP_BSEC_MIRROR_SIZE, \
 					MT_RO_DATA | MT_NS)
+#define MAP_SCMI_SEC_SHMEM	MAP_REGION_FLAT(STM32MP_SCMI_SEC_SHMEM_BASE, \
+						STM32MP_SCMI_SEC_SHMEM_SIZE, \
+						MT_DEVICE | \
+						MT_RW | \
+						MT_SECURE | \
+						MT_EXECUTE_NEVER)
 #else
 #define MAP_BSEC_MIRROR	MAP_REGION_FLAT(STM32MP_BSEC_MIRROR_BASE, \
 					STM32MP_BSEC_MIRROR_SIZE, \
@@ -98,6 +104,9 @@ static const mmap_region_t stm32mp2_mmap[] = {
 static const mmap_region_t stm32mp2_mmap[] = {
 	MAP_SEC_SYSRAM,
 	MAP_BSEC_MIRROR,
+#if STM32MP_M33_TDCID
+	MAP_SCMI_SEC_SHMEM,
+#endif /* STM32MP_M33_TDCID */
 	MAP_DEVICE,
 	{0}
 };
-- 
2.47.2

